spec/pdfs/distribution_pdf_spec.rb

```diff
@@ -79,56 +125,126 @@
+      before(:each) do
+        create(:partner_profile,
+          partner_id: partner.id,
+          primary_contact_name: profile_name,
+          primary_contact_email: profile_email,
+          address1: "",
+          address2: "",
+          city: "",
+          state: "",
+          zip_code: "",
+          program_address1: "",
+          program_address2: "",
+          program_city: "",
+          program_state: "",
+          program_zip_code: "")
+      end
+      it "doesn't print any address if the delivery type is pickup" do
+        compare_pdf(create_dist(:pick_up), expected_pickup_file)
+      end
+      it "doesn't print any address if the delivery type is delivery" do
+        compare_pdf(create_dist(:delivery), expected_pickup_file)
+      end
+      it "doesn't print any address if the delivery type is shipped" do
+        compare_pdf(create_dist(:shipped), expected_pickup_file)
+      end
     end
-  end
-  context "regardles of request data" do
-    describe "#hide_columns" do
-      it "hides package column when true on organization" do
-        pdf = described_class.new(org_hiding_packages, distribution)
-        data = pdf.request_data
-        pdf.hide_columns(data)
-        expect(data).to eq([
-          ["Items Received", "Requested", "Received", "Value/item", "In-Kind Value Received"],
-          ["Item 1", "", 50, "$1.00", "$50.00"],
-          ["Item 2", 30, 100, "$2.00", "$200.00"],
-          ["Item 3", 50, "", "$3.00", nil],
-          ["Item 4", 120, "", "$4.00", nil],
-          ["", "", "", "", ""],
-          ["Total Items Received", 200, 150, "", "$250.00"]
-        ])
+    context "when the partner doesn't have a different program address" do
+      before(:each) do
+        create_profile_without_program_address
+      end
+      it "prints the address if the delivery type is delivery" do
+        compare_pdf(create_dist(:delivery), expected_same_address_file)
+      end
+      it "prints the address if the delivery type is shipped" do
+        compare_pdf(create_dist(:shipped), expected_same_address_file)
+      end
+      it "doesn't print the address if the delivery type is pickup" do
+        compare_pdf(create_dist(:pick_up), expected_pickup_file)
+      end
+    end
+    context "when the partner has a different program/delivery address" do
+      before(:each) do
+        create_profile_with_program_address
+      end
+      it "prints the delivery address if the delivery type is delivery" do
+        compare_pdf(create_dist(:delivery), expected_different_address_file)
+      end
+      it "prints the delivery address if the delivery type is shipped" do
+        compare_pdf(create_dist(:shipped), expected_different_address_file)
+      end
+      it "doesn't print any address if the delivery type is pickup" do
+        compare_pdf(create_dist(:pick_up), expected_pickup_file)
+      end
+    end
+    # this test is a helper function to regenerate expected PDFs, only commit with it skipped
```

Comment:
That's really confusing. If you want a helper method, why not define it as a method, and people can run it from the Rails console?

Reply:
to generate the pdf files correctly, the helper method generating the file needs to have the same test environment setup as the actual test itself. I followed the same pattern recommended. This way you don't have to repeat code setting up the test environment.

you can call this method from the terminal by calling the test's line number e.g. `bundle exec rspec spec/pdfs/distribution_pdf_spec.rb:227`. I added a comment clarifying that. 

I did research but I could not figure out a better way to do it without duplicating code.

Reply:
There should definitely be a way to do this without duplicating code. Maybe it's the setup that needs to be extracted to a base method that's called from both places. And worse comes to worst, if you do need to duplicate a few lines of code, it's still a lot less messy than a test that's not actually a test.

Reply:
There should definitely be a way to do this without duplicating code. Maybe it's the setup that needs to be extracted to a base method that's called from both places. And worse comes to worst, if you do need to duplicate a few lines of code, it's still a lot less messy than a test that's not actually a test.

Reply:
I came up with a solution which checks for an environment variable and if it's set to true to overwrite the expected PDFs in [4516951](https://github.com/rubyforgood/human-essentials/pull/4540/commits/45169512a1d4a7f0e01f1ea34abda4ceceb6a659) . I think it's a lot cleaner, let me know if that works.